interceptor/reflection.go

```diff
@@ -112,21 +92,129 @@ func visitNamespace(obj any, match matcher) (bool, error) {
+			if err != nil {
+				return visit.Stop, err
+			}
+		} else if searchAttributeFieldNames[fieldType.Name] {
+			fmt.Printf("match SA field %v\n", fieldType.Name)
+			// This could be *common.SearchAttributes, or it could be map[string]*common.Payload (indexed fields)
+			var changed bool
+			fmt.Printf("before %+v\n", vwp.Interface())
+			switch attrs := vwp.Interface().(type) {
+			case *common.SearchAttributes:
+				fmt.Println("match common.SearchAttributes")
+				attrs.IndexedFields, changed = translateIndexedFields(attrs.IndexedFields, match)
+			case map[string]*common.Payload:
+				fmt.Println("match map[string]common.Payload")
+				attrs, changed = translateIndexedFields(attrs, match)
+				if changed {
+					visit.Assign(vwp, reflect.ValueOf(attrs))
+				}
+			}
+			fmt.Printf("changed=%v\n", changed)
+			fmt.Printf("after %+v\n", vwp.Interface())
+			matched = matched || changed
+
+			// No need to descend into this type further.
+			return visit.Continue, nil
+		}
+
+		return visit.Continue, nil
+	})
+	return matched, err
+}
+
+func translateIndexedFields(fields map[string]*common.Payload, match matcher) (map[string]*common.Payload, bool) {
+	if fields == nil {
+		return fields, false
+	}
+
+	var anyMatched bool
+	newIndexed := make(map[string]*common.Payload, len(fields))
+	for key, value := range fields {
+		newKey, matched := match(key)
+		anyMatched = anyMatched || matched
+		if matched && key != newKey {
+			newIndexed[newKey] = value
+		} else {
+			newIndexed[key] = value
+		}
+	}
+	return newIndexed, anyMatched
+}
+
+func getParentFieldType(vwp visit.ValueWithParent) (result reflect.StructField, action visit.Action) {
+	if vwp.Parent == nil || vwp.Parent.Kind() != reflect.Struct {
+		return result, visit.Continue
+	}
+	fieldType := vwp.Parent.Type().Field(int(vwp.Index.Int()))
+	if !fieldType.IsExported() {
+		return result, visit.Skip
+	}
+	return fieldType, action
+}
+
+func visitDataBlobs(vwp visit.ValueWithParent, match matcher, visitor visitor) (bool, error) {
+	switch evt := vwp.Interface().(type) {
+	case []*common.DataBlob:
+		newEvts, matched, err := translateDataBlobs(match, visitor, evt...)
+		if err != nil {
+			return matched, err
+		}
+		if matched {
+			if err := visit.Assign(vwp, reflect.ValueOf(newEvts)); err != nil {
+				return matched, err
+			}
+		}
+		return matched, nil
+	case *common.DataBlob:
+		newEvt, matched, err := translateOneDataBlob(match, visitor, evt)
+		if err != nil {
+			return matched, err
+		}
+		if matched {
+			if err := visit.Assign(vwp, reflect.ValueOf(newEvt)); err != nil {
+				return matched, err
+			}
+		}
+		return matched, nil
+	default:
+		return false, nil
+	}
+}
+
+func translateOneDataBlob(match matcher, visitor visitor, blob *common.DataBlob) (*common.DataBlob, bool, error) {
 	if blob == nil || len(blob.Data) == 0 {
 		return blob, false, nil
 	}
-	blobs, changed, err := translateDataBlobs(match, blob)
+	blobs, changed, err := translateDataBlobs(match, visitor, blob)
```

Comment:
would it be better to do it in the other way around:

`translateDataBlobs` calls `translateOneDataBlob` for every element? 

one potential benefit is that no need to create a new array object for every translateOneDataBlob call (potential use less memory?)